htdocs/fichinter/card-rec.php

```diff
@@ -763,22 +788,22 @@
 		}
 		$sql .= " WHERE f.fk_soc = s.rowid";
 		$sql .= " AND f.entity = ".$conf->entity;
-		if ($socid) {
+		if (!empty($socid)) {
 			$sql .= " AND s.rowid = ".((int) $socid);
 		}
 		if (empty($user->rights->societe->client->voir) && !$socid) {
 			$sql .= " AND s.rowid = sc.fk_soc AND sc.fk_user = ".((int) $user->id);
 		}
-		if ($search_ref) {
+		if (!empty($search_ref)) {
 			$sql .= natural_search('f.titre', $search_ref);
 		}
-		if ($search_societe) {
+		if (!empty($search_societe)) {
 			$sql .= natural_search('s.nom', $search_societe);
 		}
-		if ($search_frequency == '1') {
+		if (!empty($search_frequency) && $search_frequency == '1') {
 			$sql .= ' AND f.frequency > 0';
 		}
-		if ($search_frequency == '0') {
+		if (!empty($search_frequency) && $search_frequency == '0') {
```

Comment:
This test will always be wrong.   empty('0') return true 

Try
`if (isset($search_frequency) && (string) $search_frequency == '0') {`